app/web_framework/route_node.py

```diff
@@ -5,21 +5,23 @@ class RouteNode(object):
     """
     A tree data structure for representing the parts of a url path, for routing.
     """
-    def __init__(self, regex_part, handler, label=None):
+    def __init__(self, regex_part, handler, label=None, version=None):
         """
         :param regex_part: To generate the regex the web framework will use to match this route, we combine a set of
         regex_parts: The regex_part in this node and the regex_parts in all its ancestor nodes.
         :type regex_part: str
         :param handler: The handler class that will be instantiated by the web framework when this route is hit.
         :type handler: type
         :param label: A human-friendly label for the objects returned by this route, ie "builds"
+        :param version: The API version assigned to this route
```

Comment:
Can you add a type hint (might as well convert the docstring hints to annotation style while you're here).